lib/elixir/lib/calendar/datetime.ex

```diff
@@ -245,6 +362,75 @@ defmodule DateTime do
     end
   end
 
+  @doc """
+  Takes a DateTime and a time zone.
+
+  Returns a DateTime for the same point in time, but instead at the
+  time zone provided.
+
+      iex> cph_datetime = DateTime.from_naive!(~N[2018-07-16 12:00:00], "Europe/Copenhagen", FakeTimeZoneDatabase)
+      iex> {:ok, pacific_datetime} = DateTime.shift_zone(cph_datetime, "America/Los_Angeles", FakeTimeZoneDatabase)
+      iex> pacific_datetime
+      #DateTime<2018-07-16 03:00:00-07:00 PDT America/Los_Angeles>
+  """
+  @spec shift_zone(t, Calendar.time_zone(), TimeZoneDatabase.t() | :from_config) ::
+          {:ok, t} | {:error, :time_zone_not_found}
+  def shift_zone(
+        %{calendar: Calendar.ISO} = datetime,
+        time_zone,
+        time_zone_data_module \\ :from_config
+      ) do
+    in_utc = datetime |> to_unix |> from_unix! |> Map.put(:microsecond, datetime.microsecond)
```

Comment:
Looking at this dance `to_unix |> from_unix!`, what do you think about including a `to_utc/1`?

Reply:
Yes, this was a quick way to have something working. It can be optimised in various ways. For instance if the DateTime total offset (UTC + standard) is 0 then it does not have to be changed.